lib/archethic_web/live/chains/beacon_live.ex

```diff
@@ -12,28 +13,27 @@ defmodule ArchEthicWeb.BeaconChainLive do
   alias ArchEthicWeb.ExplorerView
   alias Phoenix.View
 
-  defp f2(date = %DateTime{}) do
+  defp list_transaction_by_date(date = %DateTime{}) do
     Enum.map(BeaconChain.list_subsets(), fn subset ->
-        b_address = Crypto.derive_beacon_chain_address(subset,date,True)
-        node_list = P2P.authorized_nodes()
-        nodes = Election.beacon_storage_nodes(subset,date,node_list)
-        %Node{network_patch: patch} = P2P.get_node_info()
-
-          {b_address, nodes, patch}
-      end)
-      |> :lists.flatten()
-      |> Task.async_stream(
-        fn {address, nodes, patch} ->
-          BeaconSummaryHandler.download_summary(address, nodes, patch)
-        end,
-        on_timeout: :kill_task,
-        max_concurrency: 256
-      )
-      |> Enum.to_list()
-      |>:lists.flatten()
-      |> Stream.filter(&match?({:ok, {:ok, %BeaconSummary{}}}, &1))
-      |> Enum.map(fn {:ok, {:ok, %BeaconSummary{transaction_summaries: transaction_summaries}}} -> transaction_summaries end)
-      |>:lists.flatten()
+      b_address = Crypto.derive_beacon_chain_address(subset, date, True)
+      node_list = P2P.authorized_nodes()
+      nodes = Election.beacon_storage_nodes(subset, date, node_list)
+      %Node{network_patch: patch} = P2P.get_node_info()
+
+      {b_address, nodes, patch}
+    end)
+    |> Task.async_stream(
+      fn {address, nodes, patch} ->
+        BeaconSummaryHandler.download_summary(address, nodes, patch)
+      end,
+      on_timeout: :kill_task,
+      max_concurrency: 256
+    )
+    |> Stream.filter(&match?({:ok, {:ok, %BeaconSummary{}}}, &1))
+    |> Enum.map(fn {:ok, {:ok, %BeaconSummary{transaction_summaries: transaction_summaries}}} ->
```

Comment:
You can also use Stream.map or even use Stream.flat_map, to  map and flatten at once